### PR TITLE
#42938 add topologySpreadConstraints config for istiod helm chart

### PR DESCRIPTION
**Pull request by yitingdc**
**Please provide a description of this PR:**
add topologySpreadConstraints config for istiod helm chart, can be used for better placement of Istiod workloads.

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Ambient
- [ ] Configuration Infrastructure
- [ ] Docs
- [x] Installation
- [ ] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

**Please check any characteristics that apply to this pull request.**

- [ ] Does not have any [user-facing](https://github.com/istio/istio/tree/master/releasenotes#when-to-add-release-notes) changes. This may include CLI changes, API changes, behavior changes, performance improvements, etc.

### CODE REVIEW

**Comment by hzxuzhonghu** [comment]
I donot think we should put every k8s supported feature in istio

**Comment by yitingdc** [comment]
In our cases, I want to use this feature in istio helm, or when I upgrade istio, this config will be lost

**Review by howardjohn — APPROVED** [review]

**Review by jacob-delgado — APPROVED** [review]
approving since this already exists in other charts

**Review by morvencao — APPROVED** [review]

File: manifests/charts/istio-control/istio-discovery/values.yaml [hunk 1]
```diff
@@ -41,6 +41,8 @@ pilot:
   podAnnotations: {}
   serviceAnnotations: {}
 
+  topologySpreadConstraints: []
```
Comment (hanxiaop) [inline manifests/charts/istio-control/istio-discovery/values.yaml:44]: @yitingdc This should be added in the operator API as well to pass the test.
https://github.com/istio/istio/tree/master/operator/pkg/apis/istio/v1alpha1

Comment (yitingdc) [reply manifests/charts/istio-control/istio-discovery/values.yaml:44]: thanks very much，done

**Comment by Hronom** [comment]
At what version it will be?

**Comment by hanxiaop** [comment]
> At what version it will be?

1.19

**Comment by Hronom** [comment]
@hanxiaop thanks! Can't wait=) Must have parameter actually=)

**Pull request merged**